WooCommerce/Classes/Tools/BoldableTextParser.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+
+/// Describes a text that could be bolded or not.
+struct BoldableElement: Equatable {
+    let content: String
+    let isBold: Bool
+}
+
+/// Parses boldable text for further text concatenation with bold attributes.
+struct BoldableTextParser {
+    private let boldTextPattern = "\\*{2}(.*?)\\*{2}"
+
+    /// Parses the input text and returns a collection of `BoldableElement`.
+    ///
+    /// - Returns: A collection of boldable elements.
+    func parseBoldableElements(string: String) -> [BoldableElement] {
+        guard let regex = try? NSRegularExpression(pattern: boldTextPattern) else {
+            return []
```

Comment:
Nitpick: maybe can return the whole string here.

Reply:
makes sense! updated in https://github.com/woocommerce/woocommerce-ios/pull/3285/commits/83595186048b3262514e558b9ab386af72af1a61